opencontainers/distribution/reggie/client.py

```diff
@@ -222,9 +222,11 @@ def retryRequestWithAuth(self, originalRequest, originalResponse):
             .SetQueryParam("service", h.Service)
             .SetHeader("Accept", "application/json")
             .SetHeader("User-Agent", self.Config.UserAgent)
-            .SetBasicAuth(self.Config.Username, self.Config.Password)
         )
 
+        if self.Config.Username and self.Config.Password:
```

Comment:
Could you add a comment here? I know "the code speaks for itself" but I like to be verbose so it's explicitly clear. The title of the PR would work!

Reply:
done!